dipy/reconst/tests/test_ivim.py

```diff
@@ -0,0 +1,423 @@
+    assert_array_almost_equal(ivim_fit_multi.S0_predicted, ivim_params[..., 0])
+
+
+def test_perfusion_fraction():
+    """
+    Test if the `IvimFit` class returns the correct f
+    """
+    assert_array_almost_equal(ivim_fit_single.perfusion_fraction, f)
+    assert_array_almost_equal(
+        ivim_fit_multi.perfusion_fraction, ivim_params[..., 1])
+
+
+def test_D_star():
+    """
+    Test if the `IvimFit` class returns the correct D_star
+    """
+    assert_array_almost_equal(ivim_fit_single.D_star, D_star)
+    assert_array_almost_equal(ivim_fit_multi.D_star, ivim_params[..., 2])
+
+
+def test_D():
+    """
+    Test if the `IvimFit` class returns the correct D
+    """
+    assert_array_almost_equal(ivim_fit_single.D, D)
+    assert_array_almost_equal(ivim_fit_multi.D, ivim_params[..., 3])
+
+
+def test_estimate_linear_fit():
+    """
+    Test the linear estimates considering a single exponential fit.
+    """
+    data_single_exponential_D = single_exponential(S0, D, gtab.bvals)
+    assert_array_almost_equal(ivim_model.estimate_linear_fit(
+        data_single_exponential_D,
+        split_b=500.,
+        less_than=False),
+        (S0, D))
+    data_single_exponential_D_star = single_exponential(S0, D_star, gtab.bvals)
+    assert_array_almost_equal(ivim_model.estimate_linear_fit(
+        data_single_exponential_D_star,
+        split_b=100.,
+        less_than=True),
+        (S0, D_star))
+
+
+def test_estimate_f_D_star():
+    """
+    Test if the `estimate_f_D_star` returns the correct parameters after a
+    non-linear fit.
+    """
+    params_f_D = f + 0.001, D + 0.0001
+    assert_array_almost_equal(ivim_model.estimate_f_D_star(params_f_D,
+                                                           data_single, S0, D),
+                              (f, D_star))
+
+
+def test_fit_one_stage():
+    """
+    Test to check the results for the one_stage linear fit.
+    """
+    model = IvimModel(gtab, two_stage=False)
+    fit = model.fit(data_single)
+    # assert_array_almost_equal()
+    linear_fit_params = [9.88834140e+02, 1.19707191e-01, 7.91176970e-03,
+                         9.30095210e-04]
+
+    linear_fit_signal = [988.83414044, 971.77122546, 955.46786293, 939.87125905, 924.93258982,
+                         896.85182201, 870.90346447, 846.81187693, 824.34108781, 803.28900104,
+                         783.48245048, 764.77297789, 747.03322866, 669.54798887, 605.03328304,
+                         549.00852235, 499.21077611, 454.40299244, 413.83192296, 376.98072773,
+                         343.45531017]
+
+    assert_array_almost_equal(fit.model_params, linear_fit_params)
+    assert_array_almost_equal(fit.predict(gtab), linear_fit_signal)
+
+
+def test_warning():
+    """
+    Test for warning when linear fit gives parameters which are not feasible.
+    """
+    # This signal gives negative values for the parameters
+    if SCIPY_VERSION > '0.17':
```

Comment:
This is not a great way to test for the version of a package. Use `LooseVersion` instead